ocaml/typing/typecore.ml

```diff
@@ -1464,11 +1464,13 @@ let solve_Ppat_array ~refine loc env mutability expected_ty =
     | Mutable -> Predef.type_array
   in
   (* CR layouts v4: in the future we'll have arrays of other jkinds *)
```

Comment:
Update this CR to explain that the code below (which is a bit silly: the sort will always be `Sort.value`) will get generalized once we generalize `array`.